engine/src/main/java/org/destinationsol/game/Hero.java

```diff
@@ -0,0 +1,205 @@
+        }
+        return transcendentHero;
+    }
+
+    public float getAngle() {
+        return isTranscendent ? transcendentHeroShip.getAngle() : hero.getAngle();
+    }
+
+    public Shield getShield() {
+        return isTranscendent ? transcendentHeroShip.getShield() : hero.getShield();
+    }
+
+    public Armor getArmor() {
+        return isTranscendent ? transcendentHeroShip.getArmor() : hero.getArmor();
+    }
+
+    public ShipAbility getAbility() {
+        onlySolShipHero();
+        return hero.getAbility();
+    }
+
+    public float getAbilityAwait() {
+        onlySolShipHero();
+        return hero.getAbilityAwait();
+    }
+
+    public boolean canUseAbility() {
+        onlySolShipHero();
+        return hero.canUseAbility();
+    }
+
+    public Vector2 getSpd() {
+        return isTranscendent ? transcendentHero.getSpd() : hero.getSpd();
+    }
+
+    public float getAcc() {
+        onlySolShipHero();
+        return hero.getAcc();
+    }
+
+    public Vector2 getPosition() {
+        return isTranscendent ? transcendentHero.getPosition() : hero.getPosition();
+    }
+
+    public Hull getHull() {
+        onlySolShipHero();
+        return hero.getHull();
+    }
+
+    public float getLife() {
+        return isTranscendent ? transcendentHeroShip.getLife() : hero.getLife();
+    }
+
+    public float getRotAcc() {
+        onlySolShipHero();
+        return hero.getRotAcc();
+    }
+
+    public float getRotSpd() {
+        onlySolShipHero();
+        return hero.getRotSpd();
+    }
+
+    public float getMoney() {
+        return isTranscendent ? transcendentHeroShip.getMoney() : hero.getMoney();
+    }
+
+    public void setMoney(float money) {
+        onlySolShipHero();
+        hero.setMoney(money);
+    }
+
+    public TradeContainer getTradeContainer() {
+        onlySolShipHero();
+        return hero.getTradeContainer();
+    }
+
+    public ItemContainer getItemContainer() {
+        return isTranscendent ? transcendentHeroShip.getIc() : hero.getItemContainer();
+    }
+
+    public void die() {
+        isDead = true;
+    }
+
+    public boolean maybeEquip(SolGame game, SolItem item, boolean equip) {
+        onlySolShipHero();
+        return hero.maybeEquip(game, item, equip);
+    }
+
+    public boolean maybeEquip(SolGame game, SolItem item, boolean secondarySlot, boolean equip) {
+        onlySolShipHero();
+        return hero.maybeEquip(game, item, secondarySlot, equip);
+    }
+
+    public boolean maybeUnequip(SolGame game, SolItem item, boolean equip) {
+        onlySolShipHero();
+        return hero.maybeUnequip(game, item, equip);
+    }
+
+    public boolean maybeUnequip(SolGame game, SolItem item, boolean secondarySlot, boolean equip) {
+        onlySolShipHero();
+        return hero.maybeUnequip(game, item, secondarySlot, equip);
+    }
+
+    private void onlySolShipHero() {
```

Comment:
Renaming this to something like `assertSolShipHero` or `assertNonTranscendent` (preferred) shall help clarify things a bit, because the name `onlySolShipHero` does not clearly state what the method does (which is asserting that the hero is not transcendent).